### ISSUE TITLE
tests fail on 1.6-dev

### ISSUE THREAD

**Issue by AshtonSBradley** [JuliaApproximation/FastTransforms.jl#107]
This could be highly premature, but just FYI
```julia
(@v1.6) pkg> test FastTransforms
    Testing FastTransforms
Status `/private/var/folders/3g/gsxf8njs27z_2ql1xk1gb7hw0000gn/T/jl_H4ExuP/Project.toml`
  [621f4979] AbstractFFTs v0.5.0
  [b99e7846] BinaryProvider v0.5.10
  [717857b8] DSP v0.6.6
  [7a1cc6ca] FFTW v1.2.2
  [442a2c76] FastGaussQuadrature v0.4.2
  [057dd010] FastTransforms v0.8.2
  [34b6f7d7] FastTransforms_jll v0.2.13+1
  [189a3867] Reexport v0.2.0
  [276daf66] SpecialFunctions v0.10.3
  [c751599d] ToeplitzMatrices v0.6.1
  [8f399da3] Libdl
  [37e2e46d] LinearAlgebra
  [8dfed614] Test
Status `/private/var/folders/3g/gsxf8njs27z_2ql1xk1gb7hw0000gn/T/jl_H4ExuP/Manifest.toml`
  [621f4979] AbstractFFTs v0.5.0
  [b99e7846] BinaryProvider v0.5.10
  [e66e0078] CompilerSupportLibraries_jll v0.3.3+0
  [717857b8] DSP v0.6.6
  [9a962f9c] DataAPI v1.3.0
  [864edb3b] DataStructures v0.17.19
  [7a1cc6ca] FFTW v1.2.2
  [f5851436] FFTW_jll v3.3.9+5
  [442a2c76] FastGaussQuadrature v0.4.2
  [057dd010] FastTransforms v0.8.2
  [34b6f7d7] FastTransforms_jll v0.2.13+1
  [781609d7] GMP_jll v6.2.0+1
  [1d5cc7b8] IntelOpenMP_jll v2018.0.3+0
  [c8e1da08] IterTools v1.3.0
  [856f044c] MKL_jll v2020.1.216+0
  [3a97d323] MPFR_jll v4.0.2+2
  [e1d29d7a] Missings v0.4.3
  [4536629a] OpenBLAS_jll v0.3.9+4
  [efe28fd5] OpenSpecFun_jll v0.5.3+3
  [bac558e1] OrderedCollections v1.3.0
  [f27b6e38] Polynomials v0.6.1
  [3cdcf5f2] RecipesBase v0.8.0
  [189a3867] Reexport v0.2.0
  [a2af1166] SortingAlgorithms v0.3.1
  [276daf66] SpecialFunctions v0.10.3
  [2913bbd2] StatsBase v0.32.2
  [c751599d] ToeplitzMatrices v0.6.1
  [2a0f44e3] Base64
  [ade2ca70] Dates
  [8ba89e20] Distributed
  [b77e0a4c] InteractiveUtils
  [76f85450] LibGit2
  [8f399da3] Libdl
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [d6f4376e] Markdown
  [44cfe95a] Pkg
  [de0858da] Printf
  [3fa0cd96] REPL
  [9a3f8284] Random
  [ea8e919c] SHA
  [9e88b42a] Serialization
  [6462fe0b] Sockets
  [2f01184e] SparseArrays
  [10745b16] Statistics
  [8dfed614] Test
  [cf7118a7] UUIDs
  [4ec0a83e] Unicode
ERROR: LoadError: LoadError: InitError: could not load library "/Users/abradley/.julia/artifacts/0f4c6d1a6848931c1eef5b6df0a32ec2ef5c89c7/lib/libgmpxx.4.dylib"
dlopen(/Users/abradley/.julia/artifacts/0f4c6d1a6848931c1eef5b6df0a32ec2ef5c89c7/lib/libgmpxx.4.dylib, 1): Library not loaded: @rpath/libgmp.10.dylib
  Referenced from: /Users/abradley/.julia/artifacts/0f4c6d1a6848931c1eef5b6df0a32ec2ef5c89c7/lib/libgmpxx.4.dylib
  Reason: Incompatible library version: libgmpxx.4.dylib requires version 15.0.0 or later, but libgmp.dylib provides version 14.0.0
Stacktrace:
  [1] dlopen(s::String, flags::UInt32; throw_error::Bool)
    @ Base.Libc.Libdl ./libdl.jl:114
  [2] dlopen
    @ ./libdl.jl:114 [inlined]
  [3] __init__()
    @ GMP_jll ~/.julia/packages/GMP_jll/RbrIN/src/wrappers/x86_64-apple-darwin14-cxx11.jl:57
  [4] _include_from_serialized(path::String, depmods::Vector{Any})
    @ Base ./loading.jl:697
  [5] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:782
  [6] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1007
  [7] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:928
  [8] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:923
  [9] include
    @ ./Base.jl:374 [inlined]
 [10] include(x::String)
    @ MPFR_jll ~/.julia/packages/MPFR_jll/Yb5S7/src/MPFR_jll.jl:1
 [11] top-level scope
    @ ~/.julia/packages/MPFR_jll/Yb5S7/src/MPFR_jll.jl:47
 [12] include
    @ ./Base.jl:374 [inlined]
 [13] include_package_for_output(input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId,UInt64}}, uuid_tuple::Tuple{UInt64,UInt64}, source::String)
    @ Base ./loading.jl:1219
 [14] top-level scope
    @ none:1
 [15] eval
    @ ./boot.jl:340 [inlined]
 [16] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [17] top-level scope
    @ none:1
during initialization of module GMP_jll
in expression starting at /Users/abradley/.julia/packages/MPFR_jll/Yb5S7/src/wrappers/x86_64-apple-darwin14.jl:4
in expression starting at /Users/abradley/.julia/packages/MPFR_jll/Yb5S7/src/MPFR_jll.jl:1
ERROR: LoadError: LoadError: Failed to precompile MPFR_jll [3a97d323-0669-5f0c-9066-3539efd106a3] to /Users/abradley/.julia/compiled/v1.6/MPFR_jll/NBMLS_5bBdn.ji.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] compilecache(pkg::Base.PkgId, path::String)
    @ Base ./loading.jl:1350
  [3] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1030
  [4] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:928
  [5] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:923
  [6] include
    @ ./Base.jl:374 [inlined]
  [7] include(x::String)
    @ FastTransforms_jll ~/.julia/packages/FastTransforms_jll/OzQ7R/src/FastTransforms_jll.jl:1
  [8] top-level scope
    @ ~/.julia/packages/FastTransforms_jll/OzQ7R/src/FastTransforms_jll.jl:50
  [9] include
    @ ./Base.jl:374 [inlined]
 [10] include_package_for_output(input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId,UInt64}}, uuid_tuple::Tuple{UInt64,UInt64}, source::String)
    @ Base ./loading.jl:1219
 [11] top-level scope
    @ none:1
 [12] eval
    @ ./boot.jl:340 [inlined]
 [13] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [14] top-level scope
    @ none:1
in expression starting at /Users/abradley/.julia/packages/FastTransforms_jll/OzQ7R/src/wrappers/x86_64-apple-darwin14-libgfortran5.jl:6
in expression starting at /Users/abradley/.julia/packages/FastTransforms_jll/OzQ7R/src/FastTransforms_jll.jl:1
ERROR: LoadError: LoadError: Failed to precompile FastTransforms_jll [34b6f7d7-08f9-5794-9e10-3819e4c7e49a] to /Users/abradley/.julia/compiled/v1.6/FastTransforms_jll/tPLVH_5bBdn.ji.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] compilecache(pkg::Base.PkgId, path::String)
    @ Base ./loading.jl:1350
  [3] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1030
  [4] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:928
  [5] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:923
  [6] top-level scope
    @ ~/.julia/packages/FastTransforms/MjTJy/src/libfasttransforms.jl:10
  [7] include
    @ ./Base.jl:374 [inlined]
  [8] include(x::String)
    @ FastTransforms ~/.julia/packages/FastTransforms/MjTJy/src/FastTransforms.jl:1
  [9] top-level scope
    @ ~/.julia/packages/FastTransforms/MjTJy/src/FastTransforms.jl:49
 [10] include
    @ ./Base.jl:374 [inlined]
 [11] include_package_for_output(input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId,UInt64}}, uuid_tuple::Tuple{UInt64,UInt64}, source::String)
    @ Base ./loading.jl:1219
 [12] top-level scope
    @ none:1
 [13] eval
    @ ./boot.jl:340 [inlined]
 [14] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [15] top-level scope
    @ none:1
in expression starting at /Users/abradley/.julia/packages/FastTransforms/MjTJy/src/libfasttransforms.jl:2
in expression starting at /Users/abradley/.julia/packages/FastTransforms/MjTJy/src/FastTransforms.jl:1
ERROR: LoadError: Failed to precompile FastTransforms [057dd010-8810-581a-b7be-e3fc3b93f78c] to /Users/abradley/.julia/compiled/v1.6/FastTransforms/5Lm8s_5bBdn.ji.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1350
 [3] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1030
 [4] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:928
 [5] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:923
 [6] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [7] top-level scope
   @ none:6
in expression starting at /Users/abradley/.julia/packages/FastTransforms/MjTJy/test/runtests.jl:1
ERROR: Package FastTransforms errored during testing
```

**Comment by AshtonSBradley** [comment]
```julia
julia> versioninfo()
Julia Version 1.6.0-DEV.355
Commit 955beb6a91 (2020-07-02 10:28 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.4.0)
  CPU: Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 4
```

**Comment by MikaelSlevinsky** [comment]
Thanks for the report. This issue has been encountered https://github.com/JuliaApproximation/FastTransforms.jl/issues/106, though it's not clear to me precisely what will happen in Julia v1.6+. This is in fact an issue with a dependency of a dependency, so I'd anticipate that this will be fixed externally.

**Comment by AshtonSBradley** [comment]
this seems to have been fixed. tests still freeze as previously reported, but don't error

**Comment by MikaelSlevinsky** [comment]
🐠

**Comment by AshtonSBradley** [comment]
giving this a go today on MacOS and julia 1.6, gives
```julia 
     Testing Running tests...
Test Summary:     | Pass  Total
Special functions |   19     19
Test Summary:       | Pass  Total
Chebyshev transform |  358    358
Test Summary:                        | Pass  Total
Fejér and Clenshaw–Curtis quadrature |    9      9
ERROR: LoadError: LoadError: UndefVarError: set_num_threads not defined
Stacktrace:
 [1] getproperty(x::Module, f::Symbol)
   @ Base ./Base.jl:26
 [2] top-level scope
   @ ~/.julia/packages/FastTransforms/rXHu9/test/libfasttransformstests.jl:3
 [3] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [4] top-level scope
   @ ~/.julia/packages/FastTransforms/rXHu9/test/runtests.jl:6
 [5] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [6] top-level scope
   @ none:6
in expression starting at /Users/abradley/.julia/packages/FastTransforms/rXHu9/test/libfasttransformstests.jl:3
in expression starting at /Users/abradley/.julia/packages/FastTransforms/rXHu9/test/runtests.jl:6
ERROR: Package FastTransforms errored during testing
```

**Comment by MikaelSlevinsky** [comment]
Sorry, that was just me being sloppy. Should be fixed when 0.12.3 gets tagged. https://github.com/JuliaApproximation/FastTransforms.jl/commit/a46b963989a00d113f4a36222eb564b91c89921d
